### ISSUE TITLE
[Question] ModelListGP with different initialization per model

### ISSUE THREAD

**Issue by infer742** [facebook/Ax#1832]
Hi all,
I implemented a custom Botorch model which uses a ModelListGP comprised of several SingleTaskGp's.
I am using the developer API. During model initialization, I want to init each model with a different set of points. Basically a different Sobol sequence per model. How can I do this?
I already tried adding the same metric several times to get the data distributed to different models. But then all models get initialized with the same data.
Thanks!

Cheers,
Christian

**Comment by lena-kashtelyan** [comment]
Hi, @infer742! Could you tell us a bit more about your setup and about why you are looking to initialize each sub-model with different points? It will be a bit tricky to help with this without understanding better what you are trying to do.

**Comment by infer742** [comment]
Hi,
I am trying to migrate my pure Botorch implementation of Turbo-m to Ax. The idea is to have m different trust regions, each represented by a GP model.  Now, to initialize the models each trust regions needs a different set of random points. Otherwise, the trust regions would be the same and of no use. The trust regions models also all use the same objective. They only differ with respect to the part of the search space in which optimization takes place. Since the original proposed Turbo-m algorithm suggests using the lengthscales of the models to build the trust regions, a single model for each region is needed. 

My current implementation is a subclass of BotorchModel that implements custom model_constructor and gen functions. It would work, if I could maintain the points each trust region model received separately in the Xs and Ys attribute of the model.
Basically, I am asking how to set up the optimization loop and generation strategy to construct the Xs and Ys datastructures so that:
I generate m sobol sequences initially. And end up with Xs[i] = ith sobol sequence.

I kind of rambled a bit but maybe you understand what I am trying to do.
Thanks!

**Comment by infer742** [comment]
Right, makes total sense. Thank you @SebastianAment for taking the time to provide such a detailed answer!
